core/src/main/java/org/apache/iceberg/BaseRowDelta.java

```diff
@@ -45,4 +62,80 @@ public RowDelta addDeletes(DeleteFile deletes) {
     add(deletes);
     return this;
   }
+
+  @Override
+  public RowDelta validateFromSnapshot(long snapshotId) {
+    this.startingSnapshotId = snapshotId;
+    return this;
+  }
+
+  @Override
+  public RowDelta validateDeletedFiles() {
+    return validateDeletedFiles(true);
+  }
+
+  public RowDelta validateDeletedFiles(boolean shouldValidate) {
+    this.validateDeletes = shouldValidate;
+    return this;
+  }
+
+  @Override
+  public RowDelta validateDataFilesExist(Iterable<? extends CharSequence> referencedFiles) {
```

Comment:
Do we expect only file names in this API? Would it make sense to accept `DataFIles` in the future? So that we can filter out manifests based on the partition info as we do in `ManifestFilterManager`?

Reply:
We can support both if we need to later.

My expectation is that we won't have the `DataFile` information to pass back in many cases. A simple example is `DELETE FROM`. That would be implemented with a scan that also projects `_file` and `_pos`, then writes the results into delete files in parallel tasks. It would be some work to pass additional `DataFile` fields to the writer just so we could pass more information back in metadata. It could be worth it, but I think it is reasonable to start with the simpler option.